File: tests/.hooks/csharp/primary/src/Openapi/Hooks/CustomSecurityHook.cs
#nullable enable
    namespace Openapi.Hooks
{
    using System;
    using System.Net.Http;
    using System.Net.Http.Headers;
    using System.Threading;
    using System.Threading.Tasks;
    using Openapi.Models.Shared;


    public class CustomSecurityHook : IBeforeRequestHook
    {

        public async Task<HttpRequestMessage> BeforeRequestAsync(BeforeRequestContext hookCtx, HttpRequestMessage request)
        {
            await Task.CompletedTask;

            switch (hookCtx.OperationID)
            {
                case "customSchemeAppId":
                    if (hookCtx.SecuritySource == null)
                    {
                        throw new Exception("Security source is null");
                    }


                    Security? security = hookCtx.SecuritySource() as Security;
                    if (security == null)
                    {
                        throw new Exception("Security source is not of type Security");
                    }

                    if (security.CustomSchemeAppId == null)
                    {
                        throw new Exception("CustomSchemeAppID security is not defined");
                    }

                    request.Headers.Add("X-Security-App-Id", security.CustomSchemeAppId.AppId);
                    request.Headers.Add("X-Security-Secret", security.CustomSchemeAppId.Secret);
                    break;
            }

            return request;
        }
    }
}

File: tests/.hooks/go/primary/internal/hooks/customsecurityhook.go
package hooks

import (
	"errors"
	"net/http"

	"example.com/openapi-go-sdk/pkg/models/shared"
)

type CustomSecurityHook struct{}

var _ beforeRequestHook = (*CustomSecurityHook)(nil)

func (i *CustomSecurityHook) BeforeRequest(hookCtx BeforeRequestContext, req *http.Request) (*http.Request, error) {
	switch hookCtx.OperationID {
	case "customSchemeAppId":
		if hookCtx.SecuritySource == nil {
			return nil, errors.New("security source is nil")
		}
		sec, err := hookCtx.SecuritySource(req.Context())
		if err != nil {
			return nil, err
		}

		security, ok := sec.(shared.Security)
		if !ok {
			return nil, errors.New("security source is not of type Security")
		}

		if customSecurity := security.GetCustomSchemeAppID(); customSecurity == nil {
			return nil, errors.New("customSchemeAppId security is not defined")
		} else {
			req.Header.Set("X-Security-App-Id", customSecurity.AppID)
			req.Header.Set("X-Security-Secret", customSecurity.Secret)
		}
	}

	return req, nil
}
